src/psyclone/domain/nemo/transformations/nemo_outerarrayrange2loop_trans.py

```diff
@@ -161,29 +157,24 @@ def validate(self, node, options=None):
         # Am I an assignment node?
         if not isinstance(node, Assignment):
             raise TransformationError(
-                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
-                "supplied node argument should be a PSyIR Assignment, but "
-                "found '{0}'.".format(type(node).__name__))
+                f"Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                f"supplied node argument should be a PSyIR Assignment, but "
+                f"found '{type(node).__name__}'.")
 
         # Is the LHS an array reference?
-        if not isinstance(node.lhs, (ArrayReference,
-                                     ArrayOfStructuresReference)):
+        if not (isinstance(node.lhs, Reference) and node.lhs.walk(ArrayMixin)):
             raise TransformationError(
-                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
-                "supplied assignment node should have either an ArrayReference"
-                " or an ArrayOfStructuresReference node on its lhs but found "
-                "'{0}'.".format(type(node.lhs).__name__))
-        array_reference = node.lhs
+                f"Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                f"supplied assignment node should be a Reference that contains"
+                f" an ArrayMixin somewhere in the expression, but found "
```

Comment:
Make this a bit more user friendly by talking about an "array access" rather than "ArrayMixin" please.

Reply:
Done